src/main/scala/org/constellation/domain/healthcheck/HealthCheckConsensus.scala

```diff
@@ -0,0 +1,569 @@
+package org.constellation.domain.healthcheck
+
+import cats.Order
+import cats.data.NonEmptySet
+import cats.effect.{Blocker, Clock, Concurrent, ContextShift, Fiber}
+import cats.effect.concurrent.Ref
+import cats.syntax.all._
+import io.chrisdavenport.log4cats.SelfAwareStructuredLogger
+import io.chrisdavenport.log4cats.slf4j.Slf4jLogger
+import io.circe.{Codec, Decoder, Encoder}
+import io.circe.generic.semiauto.deriveCodec
+import io.circe.syntax.EncoderOps
+import org.constellation.ConstellationExecutionContext.createSemaphore
+import org.constellation.domain.healthcheck.HealthCheckConsensusManager.{
+  FetchProposalError,
+  FetchingFailed,
+  HistoricalRound,
+  SendProposalError,
+  SendingFailed
+}
+import org.constellation.domain.p2p.PeerHealthCheck.{
+  PeerAvailable,
+  PeerHealthCheckStatus,
+  PeerUnresponsive,
+  UnknownPeer
+}
+import org.constellation.infrastructure.p2p.{ClientInterpreter, PeerResponse}
+import org.constellation.p2p.PeerData
+import org.constellation.schema.{Id, NodeState}
+import org.constellation.schema.consensus.RoundId
+
+import scala.concurrent.duration.SECONDS
+import scala.util.{Random, Try}
+
+class HealthCheckConsensus[F[_]](
+  checkedId: Id,
+  ownId: Id,
+  roundId: RoundId,
+  initialRoundPeers: Map[Id, PeerData],
+  healthCheckStatus: PeerHealthCheckStatus,
+  startedAtSecond: Long,
+  val isKeepUpRound: Boolean
+)(
+  apiClient: ClientInterpreter[F],
+  unboundedBlocker: Blocker
+)(implicit F: Concurrent[F], CS: ContextShift[F], C: Clock[F]) {
+  import HealthCheckConsensus._
+
+  implicit val roundIdOrdering: Ordering[RoundId] = Ordering.by[RoundId, String](_.id)
+  implicit val roundIdOrder: Order[RoundId] = Order.fromOrdering
+  implicit val peerDataOrdering: Ordering[PeerData] = Ordering.by[PeerData, String](_.peerMetadata.id.hex)
+  implicit val peerDataOrder: Order[PeerData] = Order.fromOrdering
+
+  val logger: SelfAwareStructuredLogger[F] = Slf4jLogger.getLogger[F]
+  val fetchProposalAfter = 120L
```

Comment:
if it's a duration, can you use it explicitly? I mean - seconds or sth

Reply:
Initially I wanted something easy to work with/compare. I'm myself up for using values that are more explicit. Will look into that.